### ISSUE TITLE
Stop relying on `TableRegressionModel`

### ISSUE THREAD

**Issue by nalimilan** [jmboehm/RegressionTables.jl#128]
We're going to stop using `TableRegressionModel` in GLM.jl (https://github.com/JuliaStats/GLM.jl/pull/339) as it's really not convenient for users to have that wrapper instead of just working with `LinearModel` and `GeneralizedLinearModel`. `TableRegressionModel` will probably be deprecated soon in StatsModels.

I see that RegressionTables.jl is one of the few packages that use `TableRegressionModel`. In particular standardized coefficients are only supported for these models:
https://github.com/jmboehm/RegressionTables.jl/blob/e13df11078b6f1bb145574d55bedcfdaa6dbb1b7/src/regtable.jl#L197-L198
IIUC, you use this to detect that some models are GLM.jl objects and access their private fields. I suspect you could achieve the same result for any model (not just GLM) by calling `modelmatrix` and `response` instead. Can you confirm this? It would be too bad to lose support for GLM.jl in RegressionTables.jl.

**Comment by jmboehm** [comment]
Yes, I've been meaning for a while to write this in a more general way. In any case I'll make sure that GLM.jl will always be supported.
